### PR TITLE
chore: modernize tooling (use GitHub Actions)

### PR DESCRIPTION
**Pull request by NMinhNguyen**
**What**: Set up GitHub actions similarly to https://github.com/kentcdodds/babel-plugin-macros/commit/ccc43ec0a0d09248d2b91abcb481037828532de5

**Why**: to make sure this package is automatically built and published

**How**: by following the changes in https://github.com/kentcdodds/babel-plugin-macros/commit/ccc43ec0a0d09248d2b91abcb481037828532de5

### CODE REVIEW

File: src/macro.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ const prevalMacros: MacroHandler = function prevalMacros({
       // That's okay, we already prevaled this one on its opening element.
     } else {
       throw new Error(
-        `babel-plugin-preval/macro can only be used as tagged template expression, function call or JSX element. You tried ${referencePath.parentPath.type}.`,
```
Comment (NMinhNguyen) [inline src/macro.ts:39]: Without this I was getting

> [typecheck] src/macro.ts(39,126): error TS2531: Object is possibly 'null'.
> [typecheck] npm run typecheck --silent exited with code 1

**Review by kentcdodds — APPROVED** [review]
Cool 👍 thanks!

**Comment by kentcdodds** [comment]
Unfortunately it looks like the actions did not pass

**Comment by NMinhNguyen** [comment]
Ah that's unfortunate. Was hard to test without seeing build results but I'll try to see locally.

**Comment by mpeyper** [comment]
I came to review and found it already merged 😅

@kentcdodds I think you need to add your npm token to the project settings https://github.com/kentcdodds/babel-plugin-preval/runs/5041913011?check_suite_focus=true#step:6:47

**Pull request merged**